### DISCUSSION TITLE
API error in getStaticProps => build error?

### DISCUSSION THREAD

**Discussion by fabb** [vercel/next.js#11342] [Help]
I couldn't find anything about thrown errors in `getStaticProps` in the docs. For example when an API request fails. I hope that this would cause a build error?

If not, that could cause compiling an error into the static page undetected, being fatal on deployment. If it would cause a build error, still we would need to **not** use `try/catch` for API calls, otherwise we would run into the same issue.

https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation

**Comment by timneutkens** [comment]
Errors thrown from getStaticProps will cause the build to fail

**Reply by fabb** [reply]
Ok thx. Then we should not use `try/catch` for api requests in `getStaticProps`.
